server.js

```diff
@@ -0,0 +1,219 @@
+const express = require('express');
+const async = require('async');
+const cors = require('cors');
+const request = require('request-promise');
+const app = express();
+const bodyParser = require('body-parser');
+const fs = require('fs');
+const getContactFromApi = require('./lib/getContactFromApi');
+const getFieldSettings = require('./lib/getFieldSettings');
+const config = require('./config/index.js');
+const port = config.port;
+const mappingObject = config.mappingObject;
+
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+app.use(bodyParser.json());
+app.use(cors());
+
+app.post('/api/saved-fields', async (req, res) => {
+  let apiKey = req.body.apiKey;
+  let obj;
+  fs.readFile(`preferences/${apiKey}-preferences.json`, 'utf8', (err, data) => {
+    // if no file exists, make an empty file
+    if (err && err.code == "ENOENT") {
+      fs.writeFile(`preferences/${apiKey}-preferences.json`, "", 'utf8', (err) => {
+        if (err) {
+          res.statusCode = 500;
+          console.log(err);
+          return res.end();
+        }
+        console.log("New settings file made.");
+      });
+    } else if (err) {
+      res.statusCode = 500;
+      console.log(err);
+      return res.end();
+    }
+    if (data) {
+      obj = JSON.parse(data);
+      res.json(obj);
+    }
+  });
+});
+
+app.post('/api/custom-fields', async (req, res) => {
+  let apiKey = req.body.apiKey;
+  let customFields = [];
+
+  request({
+      url: `https://api2.autopilothq.com/v1/contacts/custom_fields`,
+      json: true,
+      headers: {
+        'Content-Type': 'application/json',
+        'autopilotapikey': apiKey
+      },
+    }, (err, response, body) => {
+      if (err) {
+        res.statusCode = 500;
+        console.log(err);
+        return res.end()
+      } else if (res.statusCode !== 200) {
+        res.statusCode = 401;
+        console.log(err);
+        return res.end()
+      } else {
+        Array.from(body).forEach ( (customField) => {
+          customFields.push(customField.name);
+        })
+      }
+    }).catch((err) => {
+      if (err.statusCode === 401) {
+        res.end()
```

Comment:
We should handle other statusCodes here or at least provide a generic error handler?

Reply:
Thanks Ned, have fixed this up.